power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_se_solver.hpp

```diff
@@ -163,38 +178,506 @@ template <bool sym> class NewtonRaphsonSESolver {
+
+                    switch (type) {
+                    case YBusElementType::shunt:
+                        if (measured_values.has_shunt(obj)) {
+                            auto const& ys = param.shunt_param[obj];
+                            auto const& measured_power = measured_values.shunt_power(obj);
+                            process_shunt_measurement(block, rhs_block, ys, u_state, measured_power);
+                        }
+                        break;
+                    case YBusElementType::bft:
+                        [[fallthrough]];
+                    case YBusElementType::btf: {
+                        auto const& y_branch = param.branch_param[obj];
+                        if (measured_values.has_branch_from(obj)) {
+                            auto const ij_voltage_order = (type == YBusElementType::bft);
+                            process_branch_measurement(block, diag_block, rhs_block, y_branch.yff(), y_branch.yft(),
+                                                       u_state, ij_voltage_order,
+                                                       measured_values.branch_from_power(obj));
+                        }
+                        if (measured_values.has_branch_to(obj)) {
+                            auto const ij_voltage_order = (type == YBusElementType::btf);
+                            process_branch_measurement(block, diag_block, rhs_block, y_branch.ytt(), y_branch.ytf(),
+                                                       u_state, ij_voltage_order, measured_values.branch_to_power(obj));
+                        }
+                        break;
+                    }
+                    default:
+                        assert(type == YBusElementType::bff || type == YBusElementType::btt);
+                        break;
+                    }
+                }
+
+                // fill block with injection measurement constraints
+                if (measured_values.has_bus_injection(row)) {
+                    auto const& yij = y_bus.admittance()[data_idx];
+                    process_injection_row(block, diag_block, rhs_block, yij, u_state);
+                    if (row == col) {
+                        process_injection_diagonal(block, rhs_block, measured_values.bus_injection(row));
+                    }
+                } else if (row == col) {
+                    virtually_remove_constraints(block);
+                }
+            }
+        }
+
+        fill_qt(y_bus);
+        process_lagrange_multiplier(y_bus);
 
         // prefactorize
         sparse_solver_.prefactorize(data_gain_, perm_);
     }
 
-    double iterate_unknown(ComplexValueVector<sym>& u) {
+    /// Q_ij = 0
+    /// R_ii = -1.0, only diagonal
+    /// assign -1.0 to diagonal of 3x3 tensor, for asym
+    void virtually_remove_constraints(NRSEGainBlock<sym>& block) {
+        block.r_P_theta() = RealTensor<sym>{-1.0};
+        block.r_Q_v() = RealTensor<sym>{-1.0};
+    }
```

Comment:
Not sure if this function actually does what's claimed: doesn't seem to distinguish `asym` from `sym` and it does not seem to do anything to diagonal

Reply:
the constructor of `RealTensor<sym>` creates a diagonal object. E.g. `1` has the same behaviour as the unit matrix. `RealTensor` by definition is a `double` for `sym == true` and a tensor for `sym == false`. this is just us making our lives easier